src/convolution.cpp

```diff
@@ -772,6 +774,16 @@ std::ostream& operator<<(std::ostream& stream, const ConvolutionDescriptor& c)
     return stream;
 }
 
+void to_json(nlohmann::json& json, const ConvolutionAttribute::Gfx90aFp16alt& attribute)
+{
+    json = {{"value", attribute.value}};
+}
```

Comment:
Just an idea, rather than bringing in a json header for every descriptor we could have a templated friend function:

```cpp
class Gfx90aFp16alt
{
public:
    template<class Json>
    friend void to_json(Json& j, const ConvolutionAttribute::Gfx90aFp16alt& attribute)
    {
        j = {{"value", attribute.value}};
    }
};
```

I think this would still be found by ADL lookup, but I haven't tested this yet. What do you think?

Reply:
I have brought only forward declaration in headers, so it should be fine and have nice autocompletion in cpps.

Reply:
Sounds good.